### ISSUE TITLE
Expected type argument for generic class Series

### ISSUE THREAD

**Issue by ibobak** [microsoft/pyright#7951]
Type: <b>Bug</b>

Get this package: https://github.com/dllllb/pytorch-lifestream
Open the code, open `ptls/preprocessing/pandas/event_time.py`, and you will get this:

![image](https://github.com/microsoft/pylance-release/assets/8807043/8e3f7ae3-8ecf-44f1-b350-7b0530b4f23f)

I believe everything is very much OK with the code, something is wrong with Pylance

Extension version: 2024.5.1
VS Code version: Code 1.89.1 (dc96b837cf6bb4af9cd736aa3af08cf8279f7685, 2024-05-07T05:16:23.416Z)
OS version: Linux x64 6.4.6-060406-generic
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) CPU E5-2696 v4 @ 2.20GHz (88 x 1197)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|1, 1, 1|
|Memory (System)|251.76GB (227.84GB free)|
|Process Argv|--crash-reporter-id 27d42247-63fb-4d9e-9cb0-87d9974843dc|
|Screen Reader|no|
|VM|50%|
|DESKTOP_SESSION|ubuntu-xorg|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu-xorg|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
tftest:31042121
vscoreces:30445986
vscod805:30301674
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pythonmypyd1:30879173
pythoncet0:30885854
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
chatpanelt:31048053
dsvsc021:30996838
9c06g630:31013171
pythoncenvpt:31049070
fchga941:31046352
pythonprt:31047983
dwnewjupyter:31046869
26j00206:31048877

```

</details>

**Comment by max-muoto** [comment]
A pandas series takes in a generic param, if you don't want to see this error, you can reduce the strictness of your Pyright settings to avoid complaining about unused generic params, or provide it. For example, maybe you want something like:

```python
def dt_to_timestamp(x: pd.Series[dt.datetime]):
    return pd.to_datetime(x).astype('datetime64[ns]').astype('int64') // 1000000000
```

Or if you don't care to make assumptions about the type in the series:

```python
def dt_to_timestamp(x: pd.Series[Any]):
    return pd.to_datetime(x).astype('datetime64[ns]').astype('int64') // 1000000000
```

**Comment by erictraut** [comment]
> Unlike MyPy, Pyright won't automatically assume `Any` in the case of a missing generic param...

Pyright _does_ default to `Any` in the case of a missing generic parameter. That behavior is mandated by the official Python typing spec. However, pyright distinguishes between an explicit `Any` (one that appears in a type annotation) and an implicit `Any` (one that is assumed by the type checker). It displays the latter as `Unknown`.

@ibobak, it appears that you have enabled the `reportMissingTypeArgument` check. This check is off by default in "basic" or "standard" type checking modes, but it enabled in "strict" type checking mode. As @max-muoto said, you can leave this disabled if you do not want pyright to flag it as an error.

In any case, pyright is working as designed here, so this is not a bug. Closing.

**Comment by ibobak** [comment]
Thank you very much for you answers.   

To be honest, I see this syntax  pd.Series[dt.datetime]   for the first time.  What is this? For me it just doesn't work.

**Comment by erictraut** [comment]
If you're not familiar with the concept of generic types, you may find [this pyright documentation](https://microsoft.github.io/pyright/#/type-concepts?id=generic-types) useful. For more in-depth information, you can refer to [this chapter of the Python typing spec](https://typing.readthedocs.io/en/latest/spec/generics.html#generics).

**Comment by max-muoto** [comment]
> Thank you very much for you answers.   
> 
> 
> 
> To be honest, I see this syntax  pd.Series[dt.datetime]   for the first time.  What is this? For me it just doesn't work. 

Are you facing a runtime error? You might need to quote the type annotation if the actual runtime implementation doesn't use generics.
